### PR TITLE
Fixed close Button in Treatment Summary

### LINKED ISSUE #6877: Close Button in Treatment Summary is Malfunctioning
**Describe the bug**
When the treatment summary is accessed through the link i.e URL, the close button is malfunctioning 
at times it is completely not working and redirected to a  new tab

**To Reproduce**

1. Go to https://care.coronasafe.in/facility/fa4942c3-dce3-477e-81b2-7c95cd18b5da/patient/b26e30ab-7557-42d5-a5fb-14f23c086964/consultation/fae6f90b-c8d9-4593-83ff-1918f7a9bafd/treatment-summary
2. Click on Close
3. See error

**Expected behavior**
It should be redirected to patient dashboard

CC: @nihal467 @rithviknishad

### PR DESCRIPTION
**Pull request by sriharsh05**
## Proposed Changes

- Fixes #6877 
- Fixed close Button in Treatment Summary.

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Facility/TreatmentSummary.tsx [hunk 1]
```diff
@@ -105,7 +108,11 @@ const TreatmentSummary = (props: any) => {
               <i className="fas fa-print mr-2"></i> Print Treatment Summary
             </button>
             <button
-              onClick={(_) => window.history.go(-1)}
+              onClick={(_) =>
+                navigate(
+                  `/facility/${facilityId}/patient/${patientId}/consultation/${consultationId}`
+                )
+              }
```
Comment (rithviknishad) [inline src/Components/Facility/TreatmentSummary.tsx]: use `goBack` from `useAppHistory` instead of useSlug + navigate

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Facility/TreatmentSummary.tsx (hunk 1 above)
Comment (sriharsh05) [reply src/Components/Facility/TreatmentSummary.tsx]: Close Button in treatment summary is not functioning only when you access the page using a link.

Try closing treatment summary using this link : https://care.ohc.network/facility/0c95c7f0-e1d2-4aff-83fa-933cef60d3a8/patient/7e8fc20f-02da-47f7-8c33-f2d8c7e52dd8/consultation/cb44a173-6fe2-452f-b2d1-6c6e2695411e/treatment-summary

 If I use `goBack` or `window.history.go(-1)` the close button is working fine in normal cases but not working when you access the page directly through the link.

Comment (rithviknishad) [reply src/Components/Facility/TreatmentSummary.tsx]: pass that url to the goBack as arg. it'll be used as a fallback if no history is present in the app.

**Comment by sriharsh05** [comment]
@rithviknishad 
Please review the changes.

**Review by rithviknishad — APPROVED** [review]

**Review by Ashesh3 — APPROVED** [review]
LGTM.

I wonder if we could have just used `..` as the fallback URL.

**Comment by nihal467** [comment]
@sriharsh05 cypress is failing, fix the issue

**Comment by sriharsh05** [comment]
@rithviknishad 
Please approve the PR.

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**